app/views/questionnaire.py

```diff
@@ -703,14 +703,19 @@ def _get_front_end_navigation(answer_store, current_location, metadata, routing_
     return None
 
 
-def get_page_title_for_location(schema, current_location):
+def _get_question_title_for_page(first_question, context):
+    return first_question['title'] if first_question.get('title') is not None else context['question_titles'][first_question['id']]
```

Comment:
I think this is a bit more readable:
```python
if first_question.get('title') is not None:
    return first_question['title']
return context['question_titles'][first_question['id']]
```
Do you explicitly want to test for None and not empty str as well?

Reply:
its tested in `test_questionnaire`